src/fromager/commands/graph.py

```diff
@@ -88,21 +93,123 @@ def to_dot(
     graph_file: str,
     output: pathlib.Path | None,
     install_only: bool,
+    overrides_only: bool,
 ):
     "Convert a graph file to a DOT file suitable to pass to graphviz."
     graph = DependencyGraph.from_file(graph_file)
     if output:
         with open(output, "w") as f:
-            write_dot(wkctx, graph, f, install_only=install_only)
+            write_dot(wkctx, graph, f, install_only=install_only, reduce=overrides_only)
     else:
-        write_dot(wkctx, graph, sys.stdout, install_only=install_only)
+        write_dot(
+            wkctx, graph, sys.stdout, install_only=install_only, reduce=overrides_only
+        )
+
+
+def has_customizations(
+    wkctx: context.WorkContext, node: DependencyNode, overridden_packages: set[str]
+) -> bool:
+    """Check if a node has any fromager customizations (overrides, settings, patches, or plugins)."""
+    if not node.canonicalized_name:
+        # This is the root node
+        return True
+
+    name = node.canonicalized_name
+    pbi = wkctx.settings.package_build_info(name)
+
+    # Check for patches
+    all_patches: PatchMap = pbi.get_all_patches()
+    if all_patches:
+        return True
+
+    # Check for plugins
+    if pbi.plugin:
+        return True
+
+    # Check for overrides/settings
+    if name in overridden_packages:
+        return True
+
+    # Check for pre-built wheels (these are also customizations)
+    if node.pre_built:
+        return True
+
+    return False
```

Comment:
Should we move this logic into PackageBuildSettings class?

Reply:
Good idea, moved.